hbp_archive.py

```
@@ -388,13 +398,98 @@ def move(self, file_path, target_directory, new_name=None, overwrite=False):
             print("Failed to move/rename the object with error: %s" % e)
 
     def delete(self, file_path):
-        """ """
+        """Delete the specified file.
+           The following parameter needs to be specified:
+
+        file_path : string
+            Path of file to be deleted.
+        """
         try:
             self.project._connection.delete_object(self.name, file_path)
             print("Successfully deleted the object")
         except ClientException as e:
             print("Failed to delete the object with error: %s" % e)
 
+    def copy_directory(self, directory_path, target_directory, new_name=None, overwrite=False):
+        """Copy a directory to the specified directory location.
+           The original tree structure of the directory will be maintained at
+           the target location. The following parameters may be specified:
+
+        directory_path : string
+            Path of directory to be copied.
+        target_directory : string
+            Path of target directory where specified directory is to be copied.
+        new_name : string, optional
+            New name to be assigned to directory.
+        overwrite : boolean, optional
+            Specify if any already existing files at target location should be
+            overwritten. If False (default value), then only non-conflicting
+            files will be copied over.
+        """
+        if directory_path[-1] != '/':
+            directory_path += '/'
+        if not new_name:
+            new_name = os.path.basename(directory_path)
+        all_files = self.list()
+        dir_files = [f for f in all_files if f.name.startswith(directory_path)]
+        if not dir_files:
+            print("Specified directory does not exist in this container!")
```

This should raise an Exception, not print a message, so that people can handle the exception in their code if they want.

---

Sure, will do.